src/services/notificationService.ts

```diff
@@ -2,10 +2,66 @@
  * This service post a notification to the Notification hub.
  */
 
+import * as azure from "azure-sb";
+import { Either, left, right } from "fp-ts/lib/Either";
+import { IResponse } from "../app";
+import { FiscalCode } from "../types/api/FiscalCode";
 import { Notification } from "../types/notification";
+import { Device } from "../types/notification";
 
 export default class NotificationService {
+  constructor(
+    private readonly hubName: string,
+    private readonly endpointOrConnectionString: string
+  ) {}
+
   public async postNotification(_: Notification): Promise<void> {
     // TODO will be implemented by https://www.pivotaltracker.com/story/show/155934439
   }
+
+  public registerDevice(
+    fiscalCode: FiscalCode,
+    device: Device
+  ): Promise<Either<Error, IResponse<string>>> {
+    const notificationHubService = azure.createNotificationHubService(
+      this.hubName,
+      this.endpointOrConnectionString
+    );
+
+    const installation = {
+      installationId: device.installationId,
+      platform: device.platform,
+      pushChannel: device.pushChannel,
+      tags: [fiscalCode],
```

Comment:
we should use the hash of the FiscalCode, I suggest to create a `FiscalCodeHash` tagged type and also define an interface for this `installation` type that requires the tag to be a `FiscalCodeHash`